### DISCUSSION TITLE
WalletConnect v2 modal won't open if certain `localStorage` keys from previous sessions are present

### DISCUSSION THREAD

**Discussion by michaelsmueller** [WalletConnect/walletconnect-monorepo#3010] [Developer Support]
In testing, we have observed that connecting with different wallets can cause `localStorage` key values to be set that prevent the WalletConnect v2 modal from opening after reconnecting.

For example, we've needed to clear the `wc@2:client:0.3//session` entry on disconnect in our dApp (https://app.maple.finance) in order to enable the Wallet Connect v2 modal to open again when connecting with a different wallet. We're also going to clear `WALLETCONNECT_DEEPLINK_CHOICE` as this has caused other issues -- attempting to load for example `fireblocks-wc://` when connecting via say Qredo.

Some of our users continue to experience this issue of the WC2 modal not opening when using multiple wallets -- Fireblocks, Qredo, Ledger -- with our dApp. Can you provide documentation or guidance on the `localStorage` entries that should be removed on disconnect?

We'd like to balance persisting some data in the user's local browser (to allow for `eagerConnect`) but that will not interfere with subsequent connection attempts made with a different wallet.

**Comment by crypblizz8** [comment]
@0xAsimetriq

**Comment by Jakkc** [comment]
Bump

**Comment by glitch-txs** [comment]
hey are you still facing this issue? could you provide if so a list of wallets + device OS?

**Reply by michaelsmueller** [reply]
@glitch-txs The issue I described occurred when attempting to connect in Chrome (on a Macbook) using Wallet Connect v2. The WC 2 modal never opened, and we were attempting to connect either Fireblocks, Qredo or MetaMask mobile.

To solve the problem we clear `wc@2:client:0.3//session` from `localStorage` in our React dApp (among other items):

```ts
const WALLETCONNECT_DEEPLINK_CHOICE = 'WALLETCONNECT_DEEPLINK_CHOICE';
const WALLETCONNECT_SESSION = 'wc@2:client:0.3//session';
const WEB3_CONNECT_CACHED_PROVIDER = 'WEB3_CONNECT_CACHED_PROVIDER';

export function clearWeb3ConnectorCache() {
  window.localStorage.removeItem(WALLETCONNECT_DEEPLINK_CHOICE);
  window.localStorage.removeItem(WALLETCONNECT_SESSION);
  window.localStorage.removeItem(WEB3_CONNECT_CACHED_PROVIDER);
}
```

However, yes some of our users have reported issues still. We are attempting to reproduce but I can't give you any more information yet.

is there any Wallet Connect documentation on the configured `localStorage` objects?

**Reply by glitch-txs** [reply]
can you confirm you're using the latest version?

**Reply by 0xApotheosis** [reply]
We are having the same issue at ShapeShift. 

If the initial pairing attempt fails, the only way to make the modal appear again is the nuke `localStorage`.

Thanks @michaelsmueller for the idea!
